config/v1/types_authentication.go

```diff
@@ -253,6 +266,106 @@ type TokenClaimMapping struct {
 	Claim string `json:"claim"`
 }
 
+type OIDCClientConfig struct {
+	// ComponentName is the name of the component that is supposed to consume this
+	// client configuration
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Required
+	// +required
+	ComponentName string `json:"componentName"`
+
+	// ComponentNamespace is the namespace of the component that is supposed to consume this
+	// client configuration
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Required
+	// +required
+	ComponentNamespace string `json:"componentNamespace"`
+
+	// ClientID is the identifier of the OIDC client from the OIDC provider
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Required
+	// +required
+	ClientID string `json:"clientID"`
+
+	// ClientSecret refers to a secret in the `openshift-config` namespace that
+	// contains the client secret in the `clientSecret` key of the `.data` field
+	ClientSecret SecretNameReference `json:"clientSecret"`
+
+	// ExtraScopes is an optional set of scopes to request tokens with.
```

Comment:
Doc an example of when I'd want to set this.  It's not clear to me why a cluster-admin would have knowledge of scopes required that a component does not.